shops/views.py

```diff
@@ -158,4 +156,47 @@ def get_context_data(self, *args, **kwargs):
         return {'weekly_sales': weekly_sales, 'weekly_overheads': weekly_overheads}
 
 
+class WeeklyImportDropView(generic.TemplateView):
+    template_name = 'shops/weekly_imports_all.html'
+
+    def get_context_data(self, **kwargs):
+
+        # Don't need this query below as the datepicker is used instead
+        dates = models.WeeklyShopSummary.objects.all().values_list(
+            'date', 
+            flat=True
+            ).distinct()
+
+        shops = models.Shop.objects.all().values_list(
+            'code', 
+            flat=True
+            ).distinct()
+
+        return {'shops': shops, 'dates': dates}
+
+
+class WeeklyImportDetailView(generic.TemplateView):
+    template_name = 'shops/partials/weekly_imports_all_details.html'
+
+    def get_context_data(self, **kwargs):
+        # Getting the closest monday
+        input_date = datetime.strptime(
+            self.request.GET['chosen_date'],
+            '%Y-%m-%d'
+            ).date()
+        monday = input_date - timedelta(days=input_date.weekday())
+        weekly_shop_summary = models.WeeklyShopSummary.objects.get(
+            shop=models.Shop.objects.get(code=self.request.GET['chosen_shop']),
+                date=monday
+            )
+
+        weekly_sales = models.WeeklySale.objects.filter(
```

Comment:
No need to get weekly_sales and weekly_overheads here. You can access those from the weekly_shop_summary:

weekly_shop_summary.weekly_sales_set.all(), etc